core/baseline-core-v1.0-psd01.md

```diff
@@ -1598,26 +1619,129 @@ The most common access management approaches are:
 
+[[R64]](#r64) Testability:
+
+Preconditions:
+
+* The BPI is set up and configured with the specific Abstraction Layer that is being tested.
+* SSO provider and identity information have been configured for the BPI.
+
+Test Steps:
+
+1. Access the BPI system with the specific Abstraction Layer being tested.
+2. Attempt to access the BPI system and log in using SSO credentials.
+3. Ensure that the SSO user's identity is correctly mapped to their corresponding BPI identity or role.
+4. Log out of the BPI system.
+
+Expected Results:
+
+1. The Abstraction Layer provides configuration options for SSO settings.
+2. SSO is successfully integrated, allowing users to log in using SSO credentials.
+3. User mapping between SSO and BPI identities is accurate.
+4. Logging out of the BPI also logs the user out of the SSO provider.
+
 #### **[R65]**	
 Abstraction Layers utilized in a BPI MUST support multi-factor authentication.
 
 *See the link here for the NIST definition adopted in this document [MFA](https://csrc.nist.gov/glossary/term/mfa).*
 
+[[R65]](#r65) Testability: 
+
+Preconditions:
+
+* The BPI is set up and configured with the specific Abstraction Layer that is being tested.
+* MFA methods and settings are available and configured within the BPI system.
+
+Test Steps:
+
+1. Access the BPI system with the specific Abstraction Layer being tested. 
+2. Attempt to enroll a user in MFA.
+3. Log in to the BPI system with MFA-enabled credentials.
+4. Test the process for MFA recovery and reset.
+
+Expected Results:
+
+1. The BPI allows configuration of MFA settings, and the Abstraction Layer supports multiple MFA methods, such as SMS, email, authenticator apps, and hardware tokens.
+2. The Abstraction Layer allows users to set up and configure multiple MFA methods. Users can enroll in MFA with at least two distinct factors.
+3. The Abstraction Layer enforces MFA during the login process. Users must successfully complete MFA to access the BPI.
+4. Users can regain access if they lose access to one of their MFA factors. Users can reset MFA settings securely.
+
 #### **[R66]**	
 Abstraction Layers utilized in a BPI MUST support hardware security modules (HSM)
 
 *This document adopts the [NIST definition](https://csrc.nist.gov/glossary/term/hardware_security_module_hsm) and for further information, refer to [[HSM](#hsm)].*
 
+[[R66]](#r66) Testability: 
+
+Preconditions:
+
+* The BPI is set up and configured with the specific Abstraction Layer that is being tested.
+* HSM devices are available and properly integrated with the Abstraction Layer.
+
+Test Steps:
+
+1. Verify that the Abstraction Layer allows for the integration of HSM devices. 
+2. Verify that cryptographic keys used within the BPI system are stored and managed securely within the HSM.
+3. Verify that the Abstraction Layer utilizes the HSM for secure data storage.
+4. Verify that HSM is used for user authentication and access control.
```

Comment:
@KCP1023 Verify how? 

How do you test that the HSM is properly integrated? e.g. test the creation of public-private key pair. 

How do you verify secure key management in the HSM? Try to access a private key n the HSM and expect a rejection of the access request.

How do you verify that the the HSM is used for secure data storage? Do you even need to test that. Are the two test above not enough?

The Abstraction layer must only be able to create keys in the HSM, rotate keys, sign a message, and not allow access to private keys and secrets.